dev/generate_protos.py

```diff
@@ -0,0 +1,204 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+import textwrap
+import tempfile
+
+cache_dir = "build/protobuf_cache"
```

Comment:
Is https://github.com/mlflow/mlflow/pull/12360#discussion_r1647501539 addressed?